modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -117,75 +136,171 @@ template<typename Net> class Params {
     };
 
-    Params<Net>& cfgInputLayers(const typename PortCfg<Net>::In &ll) {
+    /** @brief Sets sequence of CNN input layers names for inference.
+
+    The function is used to set order of CNN input layers. This order will be
+    associated to data that you provide to inference. Count of names has to match to
+    number of CNN inputs. Name is set automatically (without calling this function)
+    if CNN has one input but this doesn't prevent you from doing it yourself.
+
+    @param layer_names array that contains names of CNN input layers.
+    @return reference to object of class Params.
+    */
+    Params<Net>& cfgInputLayers(const typename PortCfg<Net>::In &layer_names) {
         desc.input_names.clear();
-        desc.input_names.reserve(ll.size());
-        std::copy(ll.begin(), ll.end(),
+        desc.input_names.reserve(layer_names.size());
+        std::copy(layer_names.begin(), layer_names.end(),
                   std::back_inserter(desc.input_names));
         return *this;
     }
 
-    Params<Net>& cfgOutputLayers(const typename PortCfg<Net>::Out &ll) {
+    /** @brief Sets sequence of CNN output layers names for inference.
+
+    The function is used to set order of output layers. This order will be
+    associated to data that you receive from inference. Name is set automatically
+    (without calling this function) if CNN has one output but this doesn't prevent
+    you from doing it yourself. Count of names has to match to number of CNN
+    outputs.
+
+    @param layer_names array that contains names of output layers.
+    @return reference to object of class Params.
+    */
+    Params<Net>& cfgOutputLayers(const typename PortCfg<Net>::Out &layer_names) {
         desc.output_names.clear();
-        desc.output_names.reserve(ll.size());
-        std::copy(ll.begin(), ll.end(),
+        desc.output_names.reserve(layer_names.size());
+        std::copy(layer_names.begin(), layer_names.end(),
                   std::back_inserter(desc.output_names));
         return *this;
     }
 
+    /** @brief Sets constant input.
+
+    The function is used to set constant input. This input has to be
+    a prepared tensor since preprocessing is disabled for this case. You should
+    provide data and name of CNN layer which will be associated with it.
+
+    @param layer_name name of CNN layer.
+    @param data cv::Mat that contains data which will be associated with CNN layer.
+    @param hint type of input (IMAGE or TENSOR).
+    @return reference to object of class Params.
+    */
     Params<Net>& constInput(const std::string &layer_name,
                             const cv::Mat &data,
                             TraitAs hint = TraitAs::TENSOR) {
         desc.const_inputs[layer_name] = {data, hint};
         return *this;
     }
 
-    Params& pluginConfig(IEConfig&& cfg) {
-        desc.config = std::move(cfg);
+    /** @brief Sets IE config.
+
+    The function is used to set configuration for device.
```

Comment:
There's no configuration for device, it is the OpenVINO plugin configuration in general.
I'd also refer here that specific options may vary from plugin to plugin.. and can be found in the OpenVINO documentation.

Reply:
Fixed.